returnn/datasets/basic.py

```diff
@@ -1260,8 +1260,11 @@ def init_dataset(kwargs, extra_kwargs=None, default_kwargs=None):
   :rtype: Dataset
   """
   assert kwargs
-  if isinstance(kwargs, Dataset):
-    return kwargs
+  if callable(kwargs):
+    # Init dataset if it was a callable/lambda returning a dataset
+    data = kwargs()
+    data.initialize()
+    return data
```

Comment:
I don't understand. You have that `if callable(kwargs)` checks already here (just directly below). Why do you add it again? This doesn't make sense.

Also, why do you remove the `if isinstance(kwargs, Dataset):`? That might break some other code which made use of this. Don't remove that.